relay-general/src/store/normalize.rs

```diff
@@ -661,6 +663,41 @@ fn normalize_logentry(logentry: &mut Annotated<LogEntry>, _meta: &mut Meta) -> P
     logentry.apply(|le, meta| logentry::normalize_logentry(le, meta))
 }
 
+// Reads device specs (family, memory, cpu, etc) from context and sets the device.class tag to high, medium, or low.
+fn normalize_device_class(event: &mut Event) {
+    let tags = &mut event.tags.value_mut().get_or_insert_with(Tags::default).0;
+    let tag_name = "device.class".to_string();
+    if let Some(ref mut contexts) = event.contexts.value_mut() {
+        if let Some(Context::Device(ref mut device)) = contexts.get_context_mut("device") {
```

Comment:
These references do not have to be mutable. But I realize there is no `Contexts::get_context` helper. Feel free to copy paste `Contexts::get_context_mut` to create an immutable version.